File: tests/output/jpa/src/main/javagen/restaurant/jpa_jdbc_resttemplate/dtos/restaurant/FactureItem.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package restaurant.jpa_jdbc_resttemplate.dtos.restaurant;

import org.springframework.data.relational.core.mapping.Column;

import jakarta.annotation.Generated;
import jakarta.validation.constraints.NotNull;

/**
 * Détail d'une facture.
 */
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
public record FactureItem(
	@Column("id") @NotNull Integer id,
	@Column("com_id") @NotNull Integer commandeId
) {
}

File: tests/output/vanilla-js/model/restaurant/facture-item.ts
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

import {DO_ID, DO_ID_2} from "../../domains";

export interface FactureItem {
    id: number;
    commandeId: number;
}

export const FactureItemEntity = {
    id: {
        type: "field",
        name: "id",
        domain: DO_ID,
        isRequired: true,
        label: "Id"
    },
    commandeId: {
        type: "field",
        name: "commandeId",
        domain: DO_ID_2,
        isRequired: true,
        label: "CommandeId"
    }
} as const;
